lib/engine/game/g_18_zoo/step/dividend.rb

```diff
@@ -5,6 +5,8 @@ module Game
     module G18ZOO
       module Step
         class Dividend < Engine::Step::Dividend
+          include Engine::Game::G18ZOO::ChooseAbilityOnOr
```

Comment:
Why does this need to be added on all the steps?  Is it not enough for the `BuyOrUsePowerOnOr` step to have this module? Or does each step in the OR actually process the power actions?

Reply:
FWIW I removed this include line from all the step classes other than `BuyOrUsePowerOnOr` and `bundle exec rspec -e ZOO` still passes

Reply:
yes, the problem is that BuyOrUsePowerOnOr is not a blocking step, so if you are for example in "buy_train" step, and you try to execute any of the "choose_ability" action, it will fail with a "choose_ability not defined inside the current step". I found the problem here:
abilities.rb
`
      def render_ability_choice_buttons
        ability_choice_buttons = @game.round.active_step.choices_ability(@selected_company).map do |choice, label|
`
this piece of code call active_step without passing the selected company, so the next method find only if active and blocking

round/base.rb
`
      def active_step(entity = nil)
        return @steps.find { |step| step.active? && step.actions(entity).any? } if entity

        @active_step ||= @steps.find { |step| step.active? && step.blocking? }
      end
`

In another PR I would like to change this to pass also the @selected_company to the active_step, but that can change the behaviour of multiple game, so I need to test it really better

Reply:
I just added also a test `sell_ticket_zoo_in_any_step` to check that choose_actions works in any step